.github/workflows/lint.yaml

```diff
@@ -0,0 +1,22 @@
+---
+name: Linters
+
+on:  # yamllint disable-line rule:truthy
+  push:
```

Comment:
If we already run for all PRs to main, what value do we gain from again running whenever a push happens? Unless we plan to push commits to the branch without introducing them as PRs?

Reply:
The idea was to allow folks to maintain a feature branch off to the side, and iterate features via successive pushes. However you make a great point, and I think this could potentially be removed. Because only our team has direct access to create branches, or directly push commits, and yet we would probably use forks anyways... so what's the point, right?

Reply:
Hmm, yeah I think I see your intent and it is valid. I guess it is an opinion thing. From my perspective, if I am debugging or something and pushing code that I know isn't perfect, but will help me figure out an issue, I don't really need to be email spammed by GitHub actions telling me some static checks failed. When I am ready, because of the `workflow_dispatch` I think I can call before I open the PR to make sure I am good to go. But it also makes sense to me to just mute the emails when you are doing debug commits and to get that feedback for each commit about whether you pass static checks regardless of if your feature is ready. I am good with either, your call.

Also I am fine with you removing the `types` part of `pull_request` again. I had the default values wrong. My apologies